### PR TITLE
Update orafce to 4.9.2

### PR DESCRIPTION
**Pull request by zhrt123**
Update orafce to the latest version for further work: to support raw type in GPDB. (raw is one of the types in Oracle, which is the same as bytea.)
- cherry-pick the previous modification for orafce to fit GPDB.
- remove decode() supported by orafce, since they have
  already been implemented in GPDB.
- enable set `protransform` in `pg_proc` (this behavior is disabled in 3.13), since the bug for setting GUC is fixed.
- enable set `typstorage` in `pg_type` with `set allow_system_table_mods=true`.

Changes with version 3.13:
- changes the storage type from `plain` to `extended` for varchar2 and nvarchar2. `plain` specifies that data of the type will always be stored in-line and not compressed. `extended` specifies that the system will first try to compress a long data value, and will move the value out of the main table row if it's still too long.
- move objects from pg_catalog/public to Oracle schema to avoid polluting these namespaces. Refer [commit](https://github.com/orafce/orafce/commit/86a1b51440ca33a04ef47fc3bb704dee26d16753)
- varchar2 substr: 
   - use oracle_substrb3 instead of bytea_substr
   - use oracle_substrb2 instead of bytea_substr_no_len
- add dbms_sql schema, which provides interfaces to use dynamic SQL to parse any DML or DDL statement using PL/SQL. 
- add regexp function
- add greatest function
- add least function
- add oracle.unistr
- add oracle.translate_oracle_modifiers
- add oracle.sys_guid
- add oracle.to_char
- add oracle.mod
- add oracle.remainder
- add dbms_utility.get_time
- guarantee function parallel safe (useless for gp)

User behavior changes:
- user needs to set search_path to oracle to use functions in orafce: ` set search_path = public, oracle;`.

Upgrade from 3.13:
- Users could still use 3.13 orafce in the latest GP7 or upgrade to 4.9 by `alter extension orafce update to '4.9'`.
- After the upgrade from 3.13, the function/type in public/pg_catalog namespace will be moved to oracle namespace. Users who want to use these functions/types without code change could set search_path like `set search_path = public, oracle`.

### CODE REVIEW

**Comment by ashwinstar** [comment]
Is this going to be upgrade-up-will-just-work story for users? Do we need to make them aware of any major changes or breakages? What gets improved with this version (mainly motivation for doing it) - having quick pointer to it in this PR itself helps (instead of needing to search through for that info). Thanks.

**Review by 50wu — APPROVED** [review]
LGTM

File: gpcontrib/orafce/utility.c
Comment (Vincy2021) [inline gpcontrib/orafce/utility.c:1]: history change 
"-#if PG_VERSION_NUM >= 130000
+#if PG_VERSION_NUM >= 120000"
is not synchronized

Comment (Vincy2021) [reply gpcontrib/orafce/utility.c:1]: mistake

**Review by Vincy2021 — APPROVED** [review]
lgtm

**Comment by zhrt123** [comment]
> Is this going to be upgrade-up-will-just-work story for users? Do we need to make them aware of any major changes or breakages?

- Users could still use 3.13 orafce in the latest GP7 or upgrade to 4.9 by `alter extension orafce update to '4.9'`.
- After the upgrade from 3.13, the function/type in public/pg_catalog namespace will be moved to oracle namespace. Users who want to use these functions/types without code change could set search_path like `set search_path = public, oracle`.

> What gets improved with this version (mainly motivation for doing it)?
- The latest version of orafce supports more functions for Oracle datatypes. and,
- changes the storage type from `plain` to `extended` for varchar2 and nvarchar2. `plain` specifies that data of the type will always be stored in-line and not compressed. `extended` specifies that the system will first try to compress a long data value, and will move the value out of the main table row if it's still too long.

**Pull request merged**